misoc/cores/vexriscv/core.py

```diff
@@ -1,18 +1,34 @@
 import os
 
 from migen import *
-
+from migen.build.platforms.sinara import kasli
 from misoc.interconnect import wishbone
 
 
 class VexRiscv(Module):
-    def __init__(self, platform, cpu_reset_address):
-        self.ibus = i = wishbone.Interface()
-        self.dbus = d = wishbone.Interface()
+    def __init__(self, platform, cpu_reset_address, with_fpu=False):
+        if isinstance(platform, kasli.Platform) and platform.hw_rev in ("v1.0", "v1.1"):
+            if with_fpu:
+                raise ValueError("VexRiscv FPU is not supported on Kasli v1.x")
+            variant = "VexRiscv_IMA"
+        elif with_fpu:
+            variant = "VexRiscv_G"
+        else:
+            variant = "VexRiscv_IMA_wide"
```

Comment:
Better let the user of VexRiscv specify the CPU variant, and keep the Kasli code in kasli.py.

Reply:
We probably need to propagate an argument from the targets (at least) to SoCCore, such that `VexRiscv_IMA` would be selected instead of `VexRiscv_IMA_wide` for Kasli v1.

Reply:
Yes

Reply:
81be86d & 239ce42 (just for simple.py)
Added `cpu_bus_width` argument to `SoCSDRAM` & `SoCCore`. It is propagated to the CPU selection part (like `cpu_type`).